src/main/java/org/truffleruby/RubyLanguage.java

```diff
@@ -226,6 +226,17 @@ protected boolean isThreadAccessAllowed(Thread thread, boolean singleThreaded) {
         }
     }
 
+    private static boolean shouldBeSingleThreaded() {
+        final String property = System.getProperty("truffleruby.single_threaded");
+
+        if (property == null) {
+            final StackTraceElement[] trace = Thread.currentThread().getStackTrace();
+            return !trace[trace.length - 1].getClassName().equals("org.truffleruby.launcher.RubyLauncher");
```

Comment:
I can't work out a better way to find out if we ran from our launcher. Classes that you might want to use are loaded in different class loaders and are protected. This might change with @pitr-ch's work on the new launcher setup. I'll resist when that's merged and that commit is in GraalVM (this PR is based on the current commit in GraalVM, and so will be an easier merge into GraalVM).

Reply:
The PR will not make it easier to access I am afraid. Could `shouldBeSingleThreaded` default to true when `System.getProperty("truffleruby.single_threaded")` is null and. Launcher can set it to `"false"`.

Reply:
Yes that was another option. I wasn't sure if SVM system properties are mutable. I'll experiment when I've merged this PR onto our tip with your new launcher. I was going to merge this PR into GraalVM without merging our latest commits in.

Reply:
You can use System.setProperty() on SVM, which seems much cleaner here.

Reply:
I'm trying the system property because the stack trace thing doesn't work on master.